sklearn/linear_model/_ransac.py

```diff
@@ -584,9 +659,19 @@ def predict(self, X):
             accept_sparse=True,
             reset=False,
         )
-        return self.estimator_.predict(X)
 
-    def score(self, X, y):
+        _raise_for_params(params, self, "predict")
+
+        if _routing_enabled():
+            predict_params = process_routing(self, "predict", **params).estimator[
+                "predict"
+            ]
+        else:
+            predict_params = dict()
```

Comment:
Now I see that we do need the else. When routing is not enabled, we have to pass an empty dict, or otherwise we get

```
>       return self.estimator_.score(X, y, **score_params)
E       UnboundLocalError: local variable 'score_params' referenced before assignment
```